src/main/java/io/boomerang/service/FlowExecutionServiceImpl.java

```diff
@@ -185,10 +181,11 @@ private void createTaskPlan(List<Task> tasks, String activityId, final Task star
     final List<String> nodes =
         GraphProcessor.createOrderedTaskList(graph, start.getTaskId(), end.getTaskId());
     final List<Task> tasksToRun = new LinkedList<>();
-    for (final String node : nodes) {
+    for (String node : nodes) {
       final Task taskToAdd =
           tasks.stream().filter(tsk -> node.equals(tsk.getTaskId())).findAny().orElse(null);
-      tasksToRun.add(taskToAdd);
+      if (taskToAdd != null)
+        tasksToRun.add(taskToAdd);
     }
```

Comment:
Hey @AndreiOla trying to understand the change here. Is the change to remove the null check? but leave null as a possibility from the filter?

Reply:
Hi @tlawrie Here I actually added the null check, so we don't end up with NPEs later, like on line 201. I think this change is already in the code from a different PR.  Anyway, the question here is: do we expect some null values? If so, we can leave it like this. If the null values are unexpected, then we need to throw some kind of exception or at least to log some error.